templates/_helpers.tpl

```diff
@@ -12,26 +12,38 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 */}}
 {{- define "concourse.web.fullname" -}}
 {{- $name := default "web" .Values.web.nameOverride -}}
+{{- if .Values.fullnameOverride -}}
+{{- printf "%s-%s" .Values.fullnameOverride $name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+{{- end -}}
 
 {{/*
 Create a default fully qualified worker node(s) name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 */}}
 {{- define "concourse.worker.fullname" -}}
 {{- $name := default "worker" .Values.worker.nameOverride -}}
+{{- if .Values.fullnameOverride -}}
+{{- printf "%s-%s" .Values.fullnameOverride $name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+{{- end -}}
 
 {{/*
 Create a default fully qualified postgresql name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 */}}
 {{- define "concourse.postgresql.fullname" -}}
 {{- $name := default "postgresql" .Values.postgresql.nameOverride -}}
+{{- if .Values.fullnameOverride -}}
+{{- printf "%s-%s" .Values.fullnameOverride $name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
```

Comment:
This part breaks the chart because the postgresql chart doesn't know we're overriding the `fullname` so it still generates a name based on the release name:
```
$ k get pods -n test
NAME                            READY   STATUS                       RESTARTS   AGE
test-postgresql-0               1/1     Running                      0          30m
thisname-web-858f65f4c9-w8zc6   0/1     CreateContainerConfigError   0          30m
thisname-worker-0               1/1     Running                      0          30m
thisname-worker-1               1/1     Running                      0          30m
```
Describing the pod shows this error:
```
Error: secret "thisname-postgresql" not found
```

In the postgresl chart:
https://github.com/bitnami/charts/blob/c9a01cb22534bc48c3f791d0454e3053554a06b8/bitnami/postgresql/values.yaml#L40-L42

Maybe changing our helper to something like this would work?:
```suggestion
{{- if .Values.postgresql.fullnameOverride -}}
{{- printf "%s" .Values.postgresql.fullnameOverride | trunc 63 | trimSuffix "-" -}}
{{- else -}}
```

The way the postgresql chart handles naming is slightly different since it only has one component; they don't append the `-postgresql` part of the name:
https://github.com/helm/charts/blob/60f4fb6379ed824d7634864d8e1a8e3b35a62f5b/stable/postgresql/templates/_helpers.tpl#L13-L21

We could add a comment in `values.yaml` as well to tell users to also change `postgresql.fullnameOverride` and that it won't have `-postgresql` automatically appended to it.

Reply:
I have always set both:
▸   ▸   --set fullnameOverride=$(NAME) \
▸   ▸   --set postgresql.fullnameOverride=$(NAME)-postgresql \
This fix is nice. Thanks